rust/src/writer/utils.rs

```diff
@@ -114,12 +111,27 @@ pub(crate) fn next_data_path(
 pub fn record_batch_from_message(
     arrow_schema: Arc<ArrowSchema>,
     message_buffer: &[Value],
-) -> Result<RecordBatch, DeltaTableError> {
-    let mut value_iter = message_buffer.iter().map(|j| Ok(j.to_owned()));
-    let options = DecoderOptions::new().with_batch_size(message_buffer.len());
-    let decoder = Decoder::new(arrow_schema, options);
+) -> DeltaResult<RecordBatch> {
+    let mut buf = vec![];
+    for message in message_buffer {
+        buf.write_all(
+            serde_json::to_string(&message)
```

Comment:
They are making you serialize the messages?! Ouch...

Reply:
Did you consider this API? https://docs.rs/arrow-json/39.0.0/arrow_json/reader/struct.Decoder.html#method.serialize

That's what Raphael built for the use case you were asking about, I think.